jupyter_client/jsonutil.py

```diff
@@ -75,7 +83,8 @@ def squash_dates(obj):
 def date_default(obj):
     """default function for packing datetime objects in JSON."""
     if isinstance(obj, datetime):
-        return obj.isoformat()
+        obj = _ensure_tzinfo(obj)
```

Comment:
AIUI, this assumes that any naive datetime is in the local timezone. Would it be reasonable to instead refuse to serialise naive datetime objects, i.e. require the calling code to make them datetime aware? There's a risk that half-smart code could use UTC timestamps without explicitly attaching the timezone info and mess things up - especially if it's written by people whose timezone aligns with UTC, and might not notice a problem.